File: 151003/Patiyuk/TaskKafka/publisher-service/src/main/java/by/bsuir/publisherservice/service/CreatorService.java
package by.bsuir.publisherservice.service;

import by.bsuir.publisherservice.dto.request.CreatorRequestTo;
import by.bsuir.publisherservice.dto.response.CreatorResponseTo;
import org.springframework.data.domain.PageRequest;

import java.util.List;

public interface CreatorService {
    List<CreatorResponseTo> getAllCreators(PageRequest pageRequest);
    CreatorResponseTo getCreatorById(Long id);
    CreatorResponseTo getCreatorByStoryId(Long id);
    CreatorResponseTo createCreator(CreatorRequestTo creator);
    CreatorResponseTo updateCreator(CreatorRequestTo creator);
    CreatorResponseTo updateCreator(Long id, CreatorRequestTo creator);
    void deleteCreator(Long id);
}

File: 151003/Alekseeva/TaskSQL/TaskSQL/Services/Implementations/CreatorService.cs
using Microsoft.EntityFrameworkCore;
using TaskSQL.Dto.Request.CreateTo;
using TaskSQL.Dto.Request.UpdateTo;
using TaskSQL.Dto.Response;
using TaskSQL.Exceptions;
using TaskSQL.Mappers;
using TaskSQL.Services.Interfaces;
using TaskSQL.Storage;

namespace TaskSQL.Services.Implementations;

public sealed class CreatorService(AppDbContext context) : ICreatorService
{
    public async Task<CreatorResponseTo> GetCreatorById(long id)
    {
        return (await context.Creators.FindAsync(id))?.ToResponse()
               ?? throw new EntityNotFoundException($"Creator with id = {id} doesn't exist.");
    }

    public async Task<IEnumerable<CreatorResponseTo>> GetCreators()
    {
        return CreatorMapper.Map(await context.Creators.ToListAsync());
    }

    public async Task<CreatorResponseTo> CreateCreator(CreateCreatorRequestTo createCreatorRequestTo)
    {
        var creator = CreatorMapper.Map(createCreatorRequestTo);
        await context.Creators.AddAsync(creator);
        await context.SaveChangesAsync();
        return CreatorMapper.Map(creator);
    }

    public async Task DeleteCreator(long id)
    {
        var creator = await context.Creators.FindAsync(id);
        if (creator == null) throw new EntityNotFoundException($"Creator with id = {id} doesn't exist.");

        context.Creators.Remove(creator);
        await context.SaveChangesAsync();
    }

    public async Task<CreatorResponseTo> UpdateCreator(UpdateCreatorRequestTo modifiedCreator)
    {
        var creator = await context.Creators.FindAsync(modifiedCreator.Id);
        if (creator == null)
            throw new EntityNotFoundException($"Creator with id = {modifiedCreator.Id} doesn't exist.");

        context.Entry(creator).State = EntityState.Modified;

        creator.Id = modifiedCreator.Id;
        creator.FirstName = modifiedCreator.FirstName;
        creator.LastName = modifiedCreator.LastName;
        creator.Login = modifiedCreator.Login;
        creator.Password = modifiedCreator.Password;

        await context.SaveChangesAsync();
        return CreatorMapper.Map(creator);
    }
}